controller/resource_test.go

```diff
@@ -473,7 +472,6 @@ func TestReconcileNamespaces(t *testing.T) {
 					}
 				}
 			} else {
-				require.Error(t, err)
```

Comment:
For simplicity, I decided to just remove the test for the a non-nil error here. The other option would have been to check the version of the Consul client and selectively test for the error if the version is < 1.15.0. This seemed a bit heavy given the scope of the changes required to support 1.15.

Reply:
Are we getting a nil error here because of a "bug" in the `api.Client` with 1.15? Maybe comment this out instead and add a TODO to circle back when the api.Client is fixed.

Reply:
I don't think that this is a "bug" because it was intentionally changed to return `404`. In which case the client consumes the `404` returns a nil error and nil resournce to indicate "not found":
1. https://github.com/hashicorp/consul/blob/main/agent/acl_endpoint.go#L169-L174
1. https://github.com/hashicorp/consul/blob/main/api/acl.go#L998-L1001
1. https://github.com/hashicorp/consul/blob/main/api/api.go#L1243-L1244

Previously this resolved to a `403`: https://github.com/hashicorp/consul/blob/v1.14.4/agent/acl_endpoint.go#L157-L160, so a non nil error was returned by the client.

Reply:
Ah. That makes sense